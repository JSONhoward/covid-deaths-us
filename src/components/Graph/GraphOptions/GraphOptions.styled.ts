import styled from 'styled-components';

export const GraphOptionsContainer = styled('div')`
position: relative;
flex-direction: column;
align-items: center;
justify-content: center;
display: flex;
height: 3rem;
width: 100%;
`

export const Button = styled('button')<{active: boolean}>`
height: 2rem;
width: 4rem;
margin: 0 2px;
border: none;
box-shadow: 1px 1px 5px rgba(0,0,0,.25);
color: ${props => props.active ? 'white' : 'black'};
background-color: ${props => props.active ? 'red' : 'none' };
cursor: pointer;
`

export const Options = styled('div')`
display: flex;
align-items: center;
justify-content: center;
flex: 1;
`

export const StatDate = styled('div')`
height: 1rem;
width: 100%;
text-align: right;
font-size: .75rem;
font-style: italic;
`