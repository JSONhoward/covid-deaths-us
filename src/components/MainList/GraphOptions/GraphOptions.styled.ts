import styled from 'styled-components';

export const GraphOptionsContainer = styled('div')`
position: relative;
display: flex;
height: 3rem;
width: 100%;
`

export const Button = styled('button')<{active: boolean}>`
height: 2rem;
width: 4rem;
margin: 0 2px;
border: none;
box-shadow: 1px 1px 5px rgba(0,0,0,.25);
color: ${props => props.active ? 'white' : 'black'};
background-color: ${props => props.active ? 'red' : 'none' };
cursor: pointer;
`

export const Dropdown = styled('select')`
display: flex;
align-items: center;
justify-content: center;
text-align: center;
height: 2rem;
border: none;
font-family: sans-serif;
font-size: 1rem;
box-shadow: 1px 1px 5px rgba(0,0,0,.25);
-moz-appearance: none;
-webkit-appearance: none;
appearance: none;
box-sizing: border-box;
margin: 5px;
padding: 0 .5rem;

&:hover {
    border: 1px solid rgba(0,0,0,.25);
    margin-left: 4px;
}

option {
    background-color: blue;
    margin: auto;
    text-align: center;
}
`

export const Options = styled('div')`
display: flex;
align-items: center;
justify-content: center;
flex: 1;
`

export const Update = styled('div')`
position: absolute;
top: 50%;
right: 0;
transform: translateY(-50%);
display: flex;
align-items: center;
justify-content: center;
flex: .5;
`