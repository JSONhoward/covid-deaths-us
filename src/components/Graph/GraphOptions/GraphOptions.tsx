import React from 'react'

import { GraphOptionsContainer, Options, Button, StatDate } from './GraphOptions.styled';

type Props = {
    handleOptions: (event: React.MouseEvent<HTMLButtonElement>) => void,
    options: string,
    dateUpdated: string
}

const GraphOptions: React.FC<Props> = ({ handleOptions, options, dateUpdated }) => {
    return (
        <GraphOptionsContainer>
            <Options>
                <p>Show Data by:</p>
                <Button data-testid='stateButton' active={options === 'state'} onClick={handleOptions} id={'state'}>State</Button>
                <Button data-testid='ageButton' active={options === 'age'} onClick={handleOptions} id={'age'}>Age</Button>
            </Options>
            <StatDate data-testid='updatedDate'>statistics updated as of {dateUpdated}</StatDate>
        </GraphOptionsContainer>
    )
}

export default GraphOptions
