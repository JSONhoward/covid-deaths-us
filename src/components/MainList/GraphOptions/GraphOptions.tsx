import React from 'react'

import { GraphOptionsContainer, Options, Button } from './GraphOptions.styled';

type Props = {
    handleOptions: (event: React.MouseEvent<HTMLButtonElement>) => void,
    options: string
}

const GraphOptions: React.FC<Props> = ({ handleOptions, options }) => {
    return (
        <GraphOptionsContainer>
            <Options>
                <p>Show Data by:</p>
                <Button active={options === 'state'} onClick={handleOptions} id={'state'}>State</Button>
                <Button active={options === 'age'} onClick={handleOptions} id={'age'}>Age</Button>
            </Options>
        </GraphOptionsContainer>
    )
}

export default GraphOptions
