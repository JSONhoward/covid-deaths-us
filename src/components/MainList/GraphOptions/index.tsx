import GraphOptions from './GraphOptions'
export default GraphOptions