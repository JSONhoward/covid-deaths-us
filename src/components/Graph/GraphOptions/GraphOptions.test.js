import { render, screen } from '@testing-library/react'
import React from 'react'
import {create} from 'react-test-renderer'
import '@testing-library/jest-dom/extend-expect'
import GraphOptions from './'

describe('GraphOptions Tests', () => {
    test('Snapshot Test', () => {
        const tree = create(<GraphOptions />)

        expect(tree).toMatchSnapshot()
    })

    test('Renders Props', () => {
        const testHandleOptions = jest.fn(() => console.log('handle options')),
        testOptions = 'test options',
        testDateUpdated = 'test date updated'

        render(<GraphOptions handleOptions={testHandleOptions} options={testOptions} dateUpdated={testDateUpdated} />)

        expect(screen.getByTestId('updatedDate')).toBeInTheDocument()
        expect(screen.getByTestId('stateButton')).toHaveAttribute('id', 'state')
        expect(screen.getByTestId('ageButton')).toHaveAttribute('id', 'age')
    })
})