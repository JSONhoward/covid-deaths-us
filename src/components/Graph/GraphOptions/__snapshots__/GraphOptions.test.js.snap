// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GraphOptions Tests Snapshot Test 1`] = `
<div
  className="sc-bdnylx cbPWbg"
>
  <div
    className="sc-dlnjPT eKCQUI"
  >
    <p>
      Show Data by:
    </p>
    <button
      className="sc-gtssRu iUOZyM"
      data-testid="stateButton"
      id="state"
    >
      State
    </button>
    <button
      className="sc-gtssRu iUOZyM"
      data-testid="ageButton"
      id="age"
    >
      Age
    </button>
  </div>
  <div
    className="sc-hKFyIo gfoLLn"
    data-testid="updatedDate"
  >
    statistics updated as of 
  </div>
</div>
`;
